src/pudl/output/pudltabl.py

```diff
@@ -1154,6 +1154,8 @@ def gens_mega_eia(
                     "retirement_date",
                     "operational_status",
                     "capacity_mw",
+                    "fuel_type_code_pudl",
+                    "planned_retirement_date",
                 ]
                 gens_cols = list(set(gens_cols + default_cols))
             self._dfs[
```

Comment:
Instead of changing the `default_cols` here and `DEFAULT_GENS_COLS` in the MCOE table something like this could happen at the top of the `gens_mega_eia` and `mcoe` functions.

```
if self._dfs["gens_mega_eia"]:
    if list of gens_cols are not included in the cached gens_mega_eia:
        update_cols = True

if update or self._dfs["gens_mega_eia"] is None or update_cols:
            recreate gens_mega_eia
```

This is less simple than just making the default columns for `gens_mega` and MCOE the same as the PPL default columns. I haven't tested whether updating the columns would require recreating the entire table if it's already cached (the inputs required are also cached?).